waku/node/v2/wakunode2.nim

```diff
@@ -194,6 +197,14 @@ proc query*(w: WakuNode, query: HistoryQuery, handler: QueryHandlerFunc) {.async
   ## Status: Implemented.
   await w.wakuStore.query(query, handler)
 
+proc filter*(w: WakuNode, filter: FilterRequest, handler: FilterHandler) {.async, gcsafe.} =
+  ## Registers for messages that match a specific filter. Triggers the handler whenever a message is received.
+  ## FilterHandler is a method that takes a MessagePush.
+  ##
+  ## Status: Implemented.
+  let id = await w.wakuFilter.subscribe(filter)
```

Comment:
Wait so filter calls wakuFilter.subscribe? This gets even more confusing. A filter function that calls wakuFilter.subscribe with a filter.

Compared with:
- A subscribe function that calls wakuFilter.subscribe with a filter, or
- A subscribeWithFilter function that calls wakuFilter.subscribe with a filter